driver-core/src/main/java/com/datastax/driver/core/ArrayBackedRow.java

```diff
@@ -239,6 +242,23 @@ public InetAddress getInet(String name) {
         return getInet(metadata.getFirstIdx(name));
     }
 
+    public Token getToken(int i) {
+        if (tokenFactory == null)
+            throw new DriverInternalError("Token factory not set. This should only happen at initialization time");
+
+        metadata.checkType(i, tokenFactory.getTokenType().getName());
+
+        ByteBuffer value = data.get(i);
+        if (value == null || value.remaining() == 0)
+            return null;
+
+        return tokenFactory.deserialize(value);
+    }
+
+    public Token getToken(String name) {
+        return getToken(metadata.getFirstIdx(name));
```

Comment:
The getter retrieves a column of a type that matches the token type for the current partitioner. The fact that it was produced with the `token()` function is a detail that should not have an impact (even though it will be the most common case and I can't think of any other meaningful use at the moment).